plugins/org.python.pydev.core/src/org/python/pydev/core/ModulesKey.java

```diff
@@ -126,16 +126,16 @@ public boolean hasPartStartingWith(final String startingWithLowerCase) {
     }
 
     public static ModulesKey fromIO(String string) {
-        List<String> split = StringUtils.split(string, '|');
+        List<String> split = StringUtils.splitKeepEmpty(string, '|');
         int size = split.size();
         if (size == 2) {
             String f = split.get(1);
             return new ModulesKey(split.get(0), f.equals("null") ? null : new File(f));
         }
-        if (size == 3) { //zipPath was empty
+        if (size == 3) { //folderPath
```

Comment:
You can't remove the logic for the zipPath here. Rather, you need to add more info in toIO so that fromIO can check that afterwards.

Reply:
> You can't remove the logic for the zipPath here. Rather, you need to add more info in toIO so that fromIO can check that afterwards.

Even with keeping empty spaces in split? (it removes cases of split.size == 3 when zipPath is empty)

Reply:
The thing is, you stopped supporting the `zipPath` use case, which would load files from the zip (so, with this change loading files from a .zip will no longer work). You need to support all 3 use cases: regular modules from files, modules loaded from .zip files and modules from folders.